### PR TITLE
Fix `ViTForMaskedImageModeling` doc example

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Same #22185

### CODE REVIEW

**Comment by amyeroberts** [comment]
@ydshieh @alaradirik The issue coming from #22152 was an oversight on my part about breaking changes. Perhaps we should revert that PR first and then agree how to introduce this change as it is intended to be added to other vision model?

**Comment by ydshieh** [comment]
Oh, I thought it was a new model head! Indeed a breaking change there. Good for me to revert, but would be nice to talk to Sylvain or Lysandre first (if you feel necessary). I will leave you judge.

Regarding a solution if we really want  to have this new attribute and the new name `MaskedImageCompletionOutput`, adding a new property (named `logits`) to `MaskedImageCompletionOutput` might be a way, but I didn't think about this deeply.

**Comment by ydshieh** [comment]
Converted to draft for now

**Comment by amyeroberts** [comment]
@ydshieh Yep - let's get @LysandreJik and @sgugger 's opinions. 

I think having the `logits` param is probably the best solution. As far as I know, it's very rare to check against the model output type itself. I believe `reconstruction` was chosen because of the `ImageSuperResolutionOutput` data class. As mentioned in the original PR - we probably do want a different model type to be returned as the documented shapes are incorrect. 

@alaradirik - could you open a PR to revert the changes?

**Comment by amyeroberts** [comment]
Actually, it's late for @alaradirik. I'll open the PR now.

**Comment by ydshieh** [comment]
Close this PR as it's clear we will and have to definitely use the original `logits`.

**Comment by alaradirik** [comment]
Sorry for being late to comment, I added the `MaskedImageCompletionOutput` to replace the inaccurate `MaskedLMOutput `class used by the masked image modeling heads (ViT and DeiT). Neither of these models have any checkpoints on the hub as mentioned in #22152 .  Swin's MIM head has its own output class but no fine-tuned checkpoints for the MIM task either.  

However, ViT and Swin's MIM head are implementations of [SimMIM](https://arxiv.org/abs/2111.09886) and SimMIM have recently released fine-tuned checkpoints for these two models (as opposed to the base model weights). I'm planning to convert these checkpoints and add a `masked-image-completion` pipeline after @sheonhan merges the ICT PR (a contemporary, better performing MIM model). It'd be great to add an output class and fix these issues. While logits is not an accurate output name in this case as the model returns full reconstructed images, I could replace `reconstruction` with `logits` and open a new PR.

What do you think @amyeroberts @sgugger?
CC @LysandreJik @ydshieh
